pyroengine/engine.py

```diff
@@ -192,7 +192,10 @@ def predict(self, frame: Image.Image, cam_id: Optional[str] = None) -> float:
 
         # Heartbeat
         if len(self.api_client) > 0 and isinstance(cam_id, str):
-            self.heartbeat(cam_id)
+            try:
+                self.heartbeat(cam_id)
+            except Exception:
```

Comment:
If we do this, we need to only catch connection error exceptions :)

Reply:
True !